internal/docker/context/context.go

```diff
@@ -0,0 +1,109 @@
+package context
+
+// The code in this file has been extracted from https://github.com/docker/cli,
+// more especifically from https://github.com/docker/cli/blob/master/cli/context/store/metadatastore.go
+// with the goal of not consuming the CLI package and all its dependencies.
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/testcontainers/testcontainers-go/internal/docker/config"
+	"github.com/testcontainers/testcontainers-go/internal/docker/context/internal"
+)
+
+const (
+	// DefaultContextName is the name reserved for the default context (config & env based)
+	DefaultContextName = "default"
+
+	// EnvOverrideContext is the name of the environment variable that can be
+	// used to override the context to use. If set, it overrides the context
+	// that's set in the CLI's configuration file, but takes no effect if the
+	// "DOCKER_HOST" env-var is set (which takes precedence.
+	EnvOverrideContext = "DOCKER_CONTEXT"
+
+	// EnvOverrideHost is the name of the environment variable that can be used
+	// to override the default host to connect to (DefaultDockerHost).
+	//
+	// This env-var is read by FromEnv and WithHostFromEnv and when set to a
+	// non-empty value, takes precedence over the default host (which is platform
+	// specific), or any host already set.
+	EnvOverrideHost = "DOCKER_HOST"
+
+	// contextsDir is the name of the directory containing the contexts
+	contextsDir = "contexts"
+
+	// metadataDir is the name of the directory containing the metadata
+	metadataDir = "meta"
+)
+
+// ErrDockerHostNotSet is the error returned when the Docker host is not set in the Docker context
+var ErrDockerHostNotSet = internal.ErrDockerHostNotSet
+
+// getContextFromEnv returns the context name from the environment variables.
+func getContextFromEnv() string {
+	if os.Getenv(EnvOverrideHost) != "" {
```

Comment:
bug: this looks like it should be an == ""?

Reply:
https://github.com/docker/cli/blob/61c6818f0b0c3caf80685b319be22f8babd11586/cli/command/cli.go#L449-L467

This is the original code from docker/cli, so it must work as is

Reply:
Interesting as that specifically breaks context if you specify a host. This is documented in the comment for `CurrentContext`. We should replicate that so folks know the history if we want to keep that behaviour as it's quite unexpected.